src/wfdlgs2.c

```diff
@@ -243,6 +243,13 @@ SearchDlgProc(register HWND hDlg, UINT wMsg, WPARAM wParam, LPARAM lParam)
 
                           SendMessage(hwndMDIClient, WM_SETREDRAW, TRUE, 0);
                           RedrawWindow(hwndMDIClient, NULL, NULL, RDW_INVALIDATE | RDW_ALLCHILDREN);
+                      } else {
+                        // Show search window immediatley
+                        SendMessage(hwndSearch, FS_CHANGEDISPLAY, CD_PATH, (LPARAM)SearchInfo.szSearch);
+                        ShowWindow(hwndSearch,
+                          GetWindowLongPtr(hwndMDIChild, GWL_STYLE) & WS_MAXIMIZE ?
+                          SW_SHOWMAXIMIZED :
+                          SW_SHOWNORMAL);
                       }

```

Comment:
This does not appear to cover all of the cases.  Based on some trials, it appears that we need to move the ShowWindow outside of the else clause and just before the break.  Then get bMaximized before the test for hwndSearch.

Reply:
5553c80 was my best guess. I detail I didn't get 'Then get bMaximized before the test for hwndSearch'.

I moved ShowWindow() out one else clause, but not direct before the break; because bMaximised would be out of scope otherwise.

I tried and played around but the dearch window always opened.
What was the use-case which caused the problems on your side?

Reply:
If the search window is open, but not the top window is the case.  In my test case, I moved the retrieval of bMaximised above if (hwndSearch) and then you can use it just before the break.

Reply:
529726c should do it.